src/decorated_surface.rs

```diff
@@ -329,23 +345,36 @@ impl<H: Handler> DecoratedSurface<H> {
     /// This string may be used to identify the surface in a task bar, window list, or other user
     /// interface elements provided by the compositor.
     pub fn set_title(&self, title: String) {
-        self.shell_surface.set_title(title);
+        match self.shell_surface {
+            shell::Surface::Xdg(ref xdg) => {
+                xdg.toplevel.set_title(title);
+            },
+            shell::Surface::Wl(ref wl) => {
+                wl.set_title(title);
+            },
+        }
     }
 
     /// Set a class for the surface.
     ///
     /// The surface class identifies the general class of applications to which the surface
     /// belongs. A common convention is to use the file name (or the full path if it is a
     /// non-standard location) of the application's .desktop file as the class.
+    ///
+    /// Note: This is ignored when using `xdg_shell`. Not sure if it should be handled.
```

Comment:
I belive the xdg_shell equivalent would be https://smithay.github.io/wayland-rs/wayland_protocols/unstable/xdg_shell/client/zxdg_toplevel_v6/struct.ZxdgToplevelV6.html#method.set_app_id